src/stcal/dark_current/dark_sub.py

```diff
@@ -86,24 +86,28 @@ def do_correction_data(science_data, dark_data, dark_output=None):
     drk_groupgap = dark_data.exp_groupgap
 
     log.info(
-        'Science data nints=%d, ngroups=%d, nframes=%d, groupgap=%d',
```

Comment:
This is the fault of the auto-formatter (both black and ruff will do the same thing), which can be turned off locally if one wants to, but I didn't review the outputs for when it did strange things.

As for the f-string, this is due to a quirk in how `python.logging` actually functions under the hood. 

When you log a message say `log.debug(msg)` whatever is in `msg` will be computed no matter what (e.g. if its an f-string or str.format string), meaning that even if the logging level is turned down so that in this case `debug` messages are not logged, there is still overhead from logging due to it still computing `msg`. However, fprintf string formatting can be written in a way that avoids this because `python.logging` passes all the arguments after the message string to the string using `%`, i.e. `log.debug("my message with %s and %s", "this", "that)`
will ultimately be rendered as `"my message %s and %s" % ("this", "that")` if the debug level logging is set, otherwise no rendering will occur.

There are other reasons why `fprintf` style string formatting is preferred by `python.logging` in addition to this. However, in any case the documentation for `python.logging` specifically says that the `fprintf` style strings should be preferred https://docs.python.org/3.12/howto/logging.html#logging-variable-data. This unfortunately is unlikely to change due to backwards compatiblity issues.

Reply:
The why is this bad? section of https://docs.astral.sh/ruff/rules/logging-string-format/#why-is-this-bad goes into further details.